docs/internal/testing/testing.md

```diff
@@ -0,0 +1,60 @@
+# Import Pathing Guidelines for Spec files
+
+## The Problem
+One of the difficult problems that we deal with as library developers is defining the API that we expose to consuming application developers. This essentially means one of two things:
+  
+  1. We have to provide all the library exports to consumers
+    * This is often a very bad idea for long term maintenance
+  2. We have to provide only a subset of the libraries' exports so that we can allow our consumers to use the parts of the library we intend them to use.
+    * Smaller API surface means lower maintenance
+
+Unfortunately, when we, the authors of Daffodil, are writing tests we are often prone to depending upon `imports` that are only available to us as library authors. This means that the tests are from a very specific perspective, our internal view. So tests that pass for us, won't necessarily pass for our consumers because they may not have an import available. Subsequently, this also means that code that actually works for in `@daffodil/demo` for example, may not work for consuming applications. 
```

Comment:
> Subsequently, this also means that code that actually works for in `@daffodil/demo` for example, may not work for consuming applications. 
I don't think `@daffodil/demo` is a great example of this because we have never imported into demo directly from a lib that I know of. I think you should use `@daffodil/cart` or some other lib instead. I also think that using demo might be a little confusing for someone who isn't as familiar with daffodil as we are. In the first sentence you write "tests are often prone to depending upon imports that are only..." but the example you give is an application that depends directly upon a lib. Would probably make it so they would need to look at source code when they ask the question: "what is demo and why can't I import it?"